### PR TITLE
feat: add support for git amend

### PR DESCRIPTION
**Pull request by whitneyit**
So as to allow users to maintain there "one commit" [style][eslint],
this PR adds the ability to specify whether or not to have the lockfile
update be amended into the previous commit.

[eslint]: https://eslint.org/docs/1.0.0/developer-guide/contributing#step-5-squash-your-commits

### CODE REVIEW

**Comment by whitneyit** [comment]
Code coverage is also :100: now!

**Comment by whitneyit** [comment]
Code coverage is also :100: now!

File: lib/update-lockfile.js [hunk 1]
```diff
@@ -47,15 +47,22 @@ module.exports = function updateLockfile (dependency, options) {
...
   exec('git add npm-shrinkwrap.json 2>/dev/null || true')
   exec('git add package-lock.json 2>/dev/null || true')
   exec('git add yarn.lock 2>/dev/null || true')
+
   exec(`git config user.email "${commitEmail}"`)
   exec(`git config user.name "${commitName}"`)
-  const updateMessage = 'chore(package): update lockfile\n\nhttps://npm.im/greenkeeper-lockfile'
-  exec(`git commit -m "${updateMessage}"`)
+
+  if (shouldNotAmend) {
```
Comment (hulkoba) [inline lib/update-lockfile.js]: Minor nit: could you swap this, so that the positive case is the default?

**Comment by whitneyit** [comment]
@hulkoba updated!

**Comment by travi** [comment]
i was excited when i saw this come through, but finally got around to giving it a shot. unfortunately, it took me longer than it should have to realize that the feature isnt included in a published release yet.

i guess i assumed this project was publishing with semantic-release already ;) any chance this could get published soon?

**Comment by whitneyit** [comment]
@hulkoba On that note, do you want me to update the docs as well?

**Comment by finnp** [comment]
@whitneyit Has this been tested? Doesn't it miss a force push in `upload.js`?

**Pull request merged**